impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -603,7 +618,14 @@ public float getHitRate(TimeUnit unit) {
             ValueHolder<V> value = fault.get();
             final OnHeapValueHolder<V> newValue;
             if(value != null) {
-              newValue = importValueFromLowerTier(key, value, now);
+              try {
+                newValue = importValueFromLowerTier(key, value, now);
```

Comment:
In case we fail here, we ideally should be able to remove the mapping from the authority. Can you try to find a way for doing this?